pydm/tests/widgets/test_enum_combo_box.py

```diff
@@ -0,0 +1,330 @@
+    1,
+    2,
+])
+def test_value_changed(qtbot, signals, value):
+    """
+    Test the widget's handling of the value changed event. In this case, it's the enum string index choice change.
+
+    Expectations:
+    The widget records the correct the enum string index change.
+
+    Parameters
+    ----------
+    qtbot : fixture
+       pytest-qt window for widget testing
+    signals : fixture
+       The signals fixture, which provides access signals to be bound to the appropriate slots
+    value : int
+       The index of the enum string to be displayed by the widget
+    """
+    pydm_enumcombobox = PyDMEnumComboBox()
+    qtbot.addWidget(pydm_enumcombobox)
+
+    signals.new_value_signal[type(value)].connect(pydm_enumcombobox.channelValueChanged)
+    signals.new_value_signal[type(value)].emit(value)
+
+    assert pydm_enumcombobox.value == value
+
+
+@pytest.mark.parametrize("index", [
+    0,
+    1,
+    -1,
+    25,
+    -25,
+])
+def test_internal_combo_box_activated_int(qtbot, signals, index):
+    """
+    Test the the capability of the widget's activated slot in sending out a new enum string index value.
+
+    Expectations:
+    The value sent out from the "activated" slot reflects the correct new index value.
+
+    Parameters
+    ----------
+    qtbot : fixture
+        pytest-qt window for widget test
+    signals : fixture
+        The signals fixture, which provides access signals to be bound to the appropriate slots
+    index : int
+        The new enum string index value
+    """
+    pydm_enumcombobox = PyDMEnumComboBox()
+    qtbot.addWidget(pydm_enumcombobox)
+
+    # Connect the send_value_signal also to the conftest's receiveValue slot to intercept and verify that the correct
+    # new index value is sent out
+    pydm_enumcombobox.send_value_signal[int].connect(signals.receiveValue)
+    pydm_enumcombobox.activated[int].emit(index)
+
+    assert signals.value == index
+
+
+# --------------------
+# NEGATIVE TEST CASES
+# --------------------
+
+@pytest.mark.parametrize("enums, expected_error_message", [
+    (None, "Invalid enum value '{0}'. The value is expected to be a valid list of string values.".format(None)),
+    ((None, "abc"), "Invalid enum type '{0}'. The expected type is 'string'.".format(type(None))),
+    ((None, 123.456), "Invalid enum type '{0}'. The expected type is 'string'".format(type(None))),
+    ((None, None, None), "Invalid enum type '{0}'. The expected type is 'string'".format(type(None))),
+    ((123,),  "Invalid enum type '{0}'. The expected type is 'string'".format(type(123))),
+    ((123.45,),  "Invalid enum type '{0}'. The expected type is 'string'".format(type(123.45))),
+    ((123, 456), "Invalid enum type '{0}'. The expected type is 'string'".format(type(123))),
+    ((123.456, None), "Invalid enum type '{0}'. The expected type is 'string'".format(type(123.456))),
+    (("spam", 123, "eggs", "ham"), "Invalid enum type '{0}'. The expected type is 'string'".format(type(123))),
```

Comment:
Regardless of what we choose to do I think this kind of testing is overboard. Be explicit in the docstring what type we expect but we don't have to handle things eloquently if the user does not follow our instructions for input

Reply:
Thanks! I just want the error messages to be as helpful as possible. As a user, I could find an error message saying "Unable to do something" as unclear, if I could be informed as I was using the wrong data type, I could try another way. Generally speaking, of course.

Reply:
I agree. However, lets imagine we want to change this error message slightly. Now we have to update this in quite a few lines. There is a tradeoff between thoroughness and making these maintainable if we do decide to slightly change behavior.

  I think what I'm getting at is the tests should select for behavior we have a reason to expect to fail. For instance do we have any reason to believe that `(123.45,)` will pass when `(123, 456)` won't when the code just checks for a `str` type?

Reply:
The reason I have a float and an int as test values is to test the error messages, not to see if either could fail with an expected str type. But I understand your point. I'll consider the tradeoff you mention in future test cases.